src/core/catalog.js

```diff
@@ -915,6 +915,22 @@ class Catalog {
     return shadow(this, "attachments", attachments);
   }
 
+  get xfaImages() {
+    const obj = this._catDict.get("Names");
+    let xfaImages = null;
+
+    if (obj instanceof Dict && obj.has("XFAImages")) {
+      const nameTree = new NameTree(obj.getRaw("XFAImages"), this.xref);
+      for (const [key, value] of nameTree.getAll()) {
+        if (!xfaImages) {
+          xfaImages = new Dict(null);
```

Comment:
Are you absolutely sure that these Streams cannot ever contain indirect objects anywhere (given all the weird things some PDF generators do)?
If not, I'd definitely suggest just using `new Dict(this.xref);` instead to avoid any future bugs (however unlikely).